src/main/java/seedu/timetable/Timetable.java

```diff
@@ -113,12 +119,77 @@ private void addModuleToList(Module module) {
         }
     }
 
+    /**
+     * Deletes the module from the list of modules taken by the user.
+     *
+     * @param module Module to be Deleted
+     * @see Module
+     */
+    public static void deleteModuleFromList(Module module) throws UniModsException {
+        String moduleCode = module.getModuleCode();
+        boolean hasModule = false;
+        for (int i = 0; i < modules.size(); i++) {
+            if (modules.get(i).getModuleCode().equals(moduleCode)) {
+                hasModule = true;
+                modules.remove(modules.get(i));
+                TextUi.printModuleDeleted(moduleCode);
+                deleteFromLessons(module);
+            }
+        }
+        if (!hasModule) {
+            throw new UniModsException(TextUi.ERROR_MODULE_NOT_FOUND);
+        }
+    }
```

Comment:
perhaps this should not be a static function because there might be different timetable objects in the future

Also consider 
 - return out of for loop once identified module has been removed from the list and lessons
 - throw exception if for loop goes through without returning, so you dont have to track a hasModule bool